Source/Request.swift

```diff
@@ -650,8 +706,8 @@ extension Request: CustomDebugStringConvertible {
 public protocol RequestDelegate: AnyObject {
     var sessionConfiguration: URLSessionConfiguration { get }
 
-    func willAttemptToRetryRequest(_ request: Request) -> Bool
-    func retryRequest(_ request: Request, ifNecessaryWithError error: Error)
+    func retryResult(for request: Request, dueTo error: Error, completion: @escaping (RetryResult?) -> Void)
```

Comment:
Why is the `RetryResult` optional in the completion handler?

Reply:
That's a great catch. For some reason when I built it, I didn't think of just having cases where the `Request` didn't have a `RequestRetrier` attached to it to just return `.doNotRetry`. For some reason I figured you'd want to get `nil` back instead. Turns out, that's silly. Much better to just return `.doNotRetry` in that case. Will fix. 👍🏻

Reply:
Removed optional in fea5fc86.